hearth/spec/hearth/http/header_list_parser_spec.rb

```diff
@@ -38,21 +72,77 @@ module Http
         end
 
         it 'parses a single item' do
-          expect(HeaderListParser.parse_http_date_list(
-                   'Mon, 16 Dec 2019 23:48:18 GMT'
-                 ))
-            .to eq([Time.parse('Mon, 16 Dec 2019 23:48:18 GMT')])
+          expect(
+            HeaderListParser.parse_http_date_list(
+              'Mon, 16 Dec 2019 23:48:18 GMT'
+            )
+          ).to eq([Time.parse('Mon, 16 Dec 2019 23:48:18 GMT')])
         end
 
         it 'parses multiple  items' do
-          expect(HeaderListParser.parse_http_date_list(
-                   'Mon, 16 Dec 2019 23:48:18 GMT,' \
-                   'Mon, 16 Dec 2019 23:48:18 GMT'
-                 ))
-            .to eq([
-                     Time.parse('Mon, 16 Dec 2019 23:48:18 GMT'),
-                     Time.parse('Mon, 16 Dec 2019 23:48:18 GMT')
-                   ])
+          expect(
+            HeaderListParser.parse_http_date_list(
+              'Mon, 16 Dec 2019 23:48:18 GMT,' \
+              'Mon, 16 Dec 2019 23:48:18 GMT'
+            )
+          ).to eq(
+            [
+              Time.parse('Mon, 16 Dec 2019 23:48:18 GMT'),
+              Time.parse('Mon, 16 Dec 2019 23:48:18 GMT')
+            ]
+          )
+        end
+      end
+
+      describe '.parse_date_time_list' do
+        it 'parses an empty list' do
+          expect(HeaderListParser.parse_date_time_list(''))
+            .to eq([])
+        end
+
+        it 'parses a single item' do
+          expect(
+            HeaderListParser.parse_date_time_list(
+              '1970-01-01T00:00:00Z'
+            )
+          ).to eq([Time.parse('1970-01-01T00:00:00Z')])
+        end
+
+        it 'parses multiple items' do
+          expect(
+            HeaderListParser.parse_date_time_list(
+              '1970-01-01T00:00:00Z,' \
+              '2000-01-02T20:34:56.123Z'
+            )
+          ).to eq(
+            [
+              Time.parse('1970-01-01T00:00:00Z'),
+              Time.parse('2000-01-02T20:34:56.123Z')
+            ]
+          )
+        end
+      end
+
+      describe '.parse_epoch_seconds_list' do
+        it 'parses an empty list' do
+          expect(HeaderListParser.parse_epoch_seconds_list(''))
+            .to eq([])
+        end
+
+        it 'parses a single item' do
+          expect(HeaderListParser.parse_epoch_seconds_list('0.0'))
+            .to eq([Time.at(0.0)])
+        end
+
+        it 'parses multiple items' do
+          expect(
+            HeaderListParser.parse_epoch_seconds_list('0.0, 946845296')
```

Comment:
The parser always converts with `.to_i`, I can't remember exactly what is specified for epoch seconds and whether it needs to be able to handle fractional seconds here or not.

Reply:
It probably doesn't hurt to do float here so we get maximum precision

Reply:
Leaving at `.to_i` per standup convo:
 * We retraced our steps to see how [TimestampFormat](https://github.com/smithy-lang/smithy-ruby/blob/8acb2a1d88719aab5f354170d180fcabbee6b494/codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/util/TimestampFormat.java) is handled everywhere else in the codegen. Note that `.to_i` is used for epoch seconds.
* Then, we looked at the smithy's protocol-tests to see if epoch seconds are covered in terms of fractional seconds: https://github.com/smithy-lang/smithy/blob/main/smithy-aws-protocol-tests/model/restJson1/fractional-seconds.smithy
* Note that only `DateTime` scenario are covered in the above test. In addition, the [documentation](https://smithy.io/2.0/spec/protocol-traits.html#timestamp-formats) is not explicitly clear on what the Deserializers should be doing for epoch-seconds (comparing to other time formats)

Based on the above findings, leaving this for now until either SmithyDocs updates with specifics or future Smithy protocol-tests starts covering this.